host-components/src/keep-loader.rs

```diff
@@ -0,0 +1,228 @@
+        kuuid, app_loader_bind_port
+    );
+    let bind_socket = format!("/tmp/enarx-keep-{}.sock", kuuid);
+    println!("binding to {}", bind_socket);
+    let keepapploader = Arc::new(Mutex::new(build_keepapploader(
+        KEEP_LOADER_STATE_UNDEF,
+        kuuid.parse().expect("problems parsing kuuid"),
+        app_loader_bind_port
+            .parse()
+            .expect("problems parsing app_loader_bind_port"),
+        "".to_string(),
+    )));
+
+    let listener = UnixListener::bind(bind_socket).unwrap();
+    //initialise state as listening
+    //TODO - error checking
+    set_state(
+        KEEP_LOADER_STATE_LISTENING,
+        //        keepapploader.lock().unwrap().clone(),
+        keepapploader.clone(),
+    );
+
+    //only one bind at a time expected here (check for auth-token?)
+    //but our connectee may drop, so keep listening
+    for mut stream in listener.incoming() {
+        match stream {
+            Ok(stream) => {
+                let childstate = keepapploader.clone();
+                thread::spawn(move || keep_loader_connection(stream, childstate));
+            }
+            Err(err) => {
+                println!("Keep-loader error in stream: {}", err);
+                //TODO - something better here, including clean-up for /tmp file
+                panic!("Stream error {}", err);
+            }
+        }
+    }
+}
+
+fn build_keepapploader(
+    state: u8,
+    kuuid: usize,
+    app_loader_bind_port: u16,
+    bindaddress: String,
+) -> KeepLoader {
+    KeepLoader {
+        state: state,
+        kuuid: kuuid,
+        app_loader_bind_port: app_loader_bind_port,
+        bindaddress: bindaddress,
+    }
+}
+
+fn keep_loader_connection(mut stream: UnixStream, keepapploader: Arc<Mutex<KeepLoader>>) {
+    let mut app_port: u16 = 0;
+    let mut json_pair: serde_json::value::Value;
+    let mut stream = &stream;
+    let deserializer = serde_json::Deserializer::from_reader(stream);
+    let iterator = deserializer.into_iter::<serde_json::Value>();
+
+    let kal = keepapploader.clone();
+
+    for json_pair in iterator {
+        match json_pair {
+            Ok(value) => {
+                let json_command: JsonCommand = serde_json::from_value(value).unwrap();
+                match json_command.commandtype.as_str() {
+                    KEEP_APP_LOADER_START_COMMAND => {
+                        //TODO - app_port may be insufficient: need to provide hostname
+                        // as well at some point.  We might also create a separate command
+                        // to provision information before starting.
+                        app_port = json_command.commandcontents.parse().unwrap();
+                        println!("About to spawn, listening on port {}", app_port.to_string());
+                        let child_spawn_result = Command::new(
+                            "/home/mike/programming/enarx/keep-runtime/target/x86_64-unknown-linux-musl/debug/keep-runtime",
```

Comment:
I've moved this to lib.rs, partly because it's going to change anyway, I presume.  I'm a little worried about putting anything in which is relative and hard-coded (particularly the "debug" part): I'm thinking this should probably be part of a build.rs script.